src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -17,6 +17,7 @@ public sealed class ResourceReader : IDisposable
         private Stream _stream;    // backing store we're reading from.
         private long _nameSectionOffset;  // Offset to name section of file.
         private long _dataSectionOffset;  // Offset to Data section of file.
+        private long _resourceStreamStart;  // beginning of the resource stream, which could be part of a longer file stream
```

Comment:
The next commit will be to actually add a metadata reader (mini) that can fish out a resource out of a larger assembly. It will be naturally testing this. But I can add a fake by shifting data in one of the existing tests.